src/application/Application.mjs

```diff
@@ -405,6 +421,187 @@ export default class Application extends Component {
         return;
     }
 
+    _recieveScrollWheel(e) {
+        const obj = e;
+        const {clientX, clientY,} = obj;
+
+        if (clientX <= this.stage.w && clientY <= this.stage.h) {
+            if (!this.fireTopDownScrollWheelHandler("_captureScroll", obj)) {
+                this.fireBottomUpScrollWheelHandler("_handleScroll", obj);
+            }
+        }
+    }
+
+    fireTopDownScrollWheelHandler(event, obj) {
+        let children = this.stage.application.children;
+        let affected = this._findChildren([], children).reverse();
+        let n = affected.length;
+
+        while(n--) {
+            const child = affected[n];
+            if (child && child[event]) {
+                child._captureScroll(obj);
+                return true; 
+            }
+        }
+        return false;
+    }
+
+    fireBottomUpScrollWheelHandler(event, obj) {
+        const {clientX, clientY} = obj;
+        const target = this._getTargetChild(clientX, clientY);
+        let child = target;
+
+        // Search tree bottom up for a handler
+        while (child !== null) {
+            if (child && child[event]) {
+                child._handleScroll(obj);
+                return true;
+            }
+            child = child.parent;
+        }
+        return false
+    }
+
+    _receiveClick(e) {
+        const obj = e;
+        const {clientX, clientY} = obj;
+
+        if (clientX <= this.stage.w && clientY <= this.stage.h) {
+            this.stage.application.fireBottomUpClickHandler(obj);
+        }
+    }
+
+    fireBottomUpClickHandler(obj) {
+        const {clientX, clientY} = obj;
+        const target = this._getTargetChild(clientX, clientY);
+        let child = target;
+
+        // Search tree bottom up for a handler
+        while (child !== null) {
+            if (child && child["_handleClick"]) {
+                child._handleClick(target);
+                break;
+            }
+            child = child.parent;
+        }
+    }
+
+    _receiveHover(e) {
+        const obj = e;
+        const {clientX, clientY} = obj;
+
+        if (clientX <= this.stage.w && clientY <= this.stage.h) {
+            this.stage.application.fireBottomUpHoverHandler(obj);
+        }
+    }
+
+    fireBottomUpHoverHandler(obj) {
+        const {clientX, clientY} = obj;
+        const target = this._getTargetChild(clientX, clientY);
+
+        // Only fire handlers when pointer target changes
+        if (target && (target !== this.__hoveredChild)) {
```

Comment:
`_handleUnhover` should fire even when new target is `null`.

Reply:
Good point, I've removed this condition.